test/ClientTest.cpp

```diff
@@ -557,6 +558,82 @@ single_conn_call(Connector<BUFFER, NetProvider> &client)
 	client.close(conn);
 }
 
+/* Authentication */
+template <class BUFFER, class NetProvider = Net_t>
+void
+authentication_test(Connector<BUFFER, NetProvider> &client)
+{
+	TEST_INIT(0);
+	TEST_CASE("Authentication within user_name and password");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == 0);
+		client.close(conn);
+	}
+	
+	TEST_CASE("Authentication with user_name and password good");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastas", "123456"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == 0);
+		client.close(conn);
+	}
+	
+	TEST_CASE("Authentication with user_name and wrong password");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastas", "1234567"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == -1);
+		client.close(conn);
+	}
+
+
+	TEST_CASE("Authentication with wrong user_name");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastasius", "123456"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == -1);
+		client.close(conn);
+	}
+
+	TEST_CASE("Authentication with some privileges without errors");
```

Comment:
I get assertion fault on this test:
Client.test: tntcxx/test/ClientTest.cpp:609: void authentication_test(Connector<BUFFER, NetProvider>&) [with BUFFER = tnt::Buffer<16384>; NetProvider = DefaultNetProvider<tnt::Buffer<16384>, NetworkEngine>]: Assertion `false' failed.